### ISSUE TITLE
`nilearn.plotting.view_img` does not handle anisotropic images correctly without a background

### ISSUE THREAD

**Issue by CarolineLndl** [nilearn/nilearn#2546]
Hi! 
I used the Nilearn tools to visualize my data on a Jupyter Notebook.
However, I have data from the spinal cord (no brain for most of my images) and when I display the spinal cord images on the background of a spinal cord anatomical image, I get a flattened image in the height direction, not the width direction. On the other hand, if I visualize this same image on an anatomical image including the brain and the spinal cord I don't have this problem.
I used interactive plotting with plotting.view_img() with Nilearn 0.6.2
Do you have any suggestions to improve this visualization of this kind of data?

Thank for your help

Caroline

**Comment by bthirion** [comment]
Hi, Thx for reporting. However, this discussion would be better place on Neurostars.
Sorry, I'm not really expert in Nilearn visualization functions, but my understanding is that you are fighting with the default parameters of the viewer. My best advice is to extract the relevant code and handle the visualization with matplotlib directly.

**Comment by CarolineLndl** [comment]
Thanks, I'll try!

**Comment by jeromedockes** [comment]
Thanks @CarolineLndl for providing more details in neurostars: https://neurostars.org/t/nilearn-display-spinal-cord-images/16240   
it seems there is indeed a problem with `view_img` whenever the image's affine isn't diagonal. @pbellec had you noticed something like this before?

**Comment by GaelVaroquaux** [comment]
We should add a call to nilearn.image.reorder_img(..., resample='continuous') at the beginning of view_img
https://nilearn.github.io/modules/generated/nilearn.image.reorder_img.html

**Comment by pbellec** [comment]
@jeromedockes yes, I recently noticed an issue with non-isotropic data, which you can reproduce here:
``` 
from nilearn import datasets
haxby_dataset = datasets.fetch_haxby()
from nilearn.plotting import view_img
view_img(haxby_dataset.anat[0], bg_img=None, cmap='gray', symmetric_cmap=False, threshold=None, vmax=250)
``` 
Here is the output. Notice the elongation along the z axis. 
![output](https://user-images.githubusercontent.com/1670887/89045947-55c7f680-d31a-11ea-942e-8cd8c7c19157.png)

Here is the expected output, which is obtained by using the default background image. 
``` 
view_img(haxby_dataset.anat[0], cmap='gray', symmetric_cmap=False, threshold=None, vmax=250)
``` 
![correct_output](https://user-images.githubusercontent.com/1670887/89046052-755f1f00-d31a-11ea-808f-e961926fdc03.png)

Brainsprite currently expect a diagonal and isotropic affine transform.

The reason it works with a background is that the overlay gets resampled on the background, and the background has an isotropic diagonal affine.. It's done here:
https://github.com/nilearn/nilearn/blob/4c1c7337b38506e6ef60873cfe3dd316ce869cf0/nilearn/plotting/html_stat_map.py#L178

The background image gets reordered, but I don't think this makes the image isotropic. It's done here:
https://github.com/nilearn/nilearn/blob/4c1c7337b38506e6ef60873cfe3dd316ce869cf0/nilearn/plotting/html_stat_map.py#L169

To fix the problem, we would need to resample the background to a diagonal, isotropic affine. Note this will automatically be applied to the overlay as well even if no background is specified, because of that block of code.  https://github.com/nilearn/nilearn/blob/4c1c7337b38506e6ef60873cfe3dd316ce869cf0/nilearn/plotting/html_stat_map.py#L165-L168

Is there an easy way to resample to isotropic resolution using nilearn?

**Comment by pbellec** [comment]
Just looking at the code of `reorder_img`, it would be fairly easy to add an option to force isotropic resampling in there. Happy to open a PR if you think it's a good idea.

**Comment by bthirion** [comment]
I'm a bit worried that we use the semantics of `reorder_img` that does not clearly involve resampling, but we certainly need this feature. And I agree that the simplest option is to add an option to `reorder_img`. Any other opinion ?

**Comment by pbellec** [comment]
another option would be to create a new function called `resample_to_self`. If it is not possible to simply reorder axes, then `reorder_img` will call `resample_to_self`. That's in effect what happens already, except the code of `resample_to_self`is mixed in `reorder_img`. Creating a new function would make the code of each function more concise, and their respective purpose clearer.

**Comment by bthirion** [comment]
+1 on my side.

**Comment by pbellec** [comment]
Quick update: the fall semester / grant season has started and I was unable to work on this. I have an opening in a few weeks and will open a PR asap.

**Comment by lunebellec** [comment]
Alright, hopefully the sprint 2026 is the time where this issue finally goes to rest. I've started collecting a few pointers to help you navigate the code @michellewang

# 1. `load_bg_img` (html\_stat\_map.py)

`load_bg_img` handles background loading (or duplicates foreground if `None`).

Within this function:

- `reorder_img` is applied.
- Comment suggests isotropic conversion.

In practice:

- Only affine reordering to XYZ is performed.
- Obliquity is removed.
- No voxel-size normalization occurs.

This is likely the core source of the bug. Possible fix 
1. Detect anisotropy via voxel sizes at the end of  `load_bg_img`:
2. If anisotropic:
   - Compute smallest voxel dimension.
   - Construct isotropic target affine.
   - Call `resample_img`.

Advantages of this approach is to have no API change even inside private functions, and likely isolated impact to `view_img` as I don't think those functions are reused in other plotting tools (but I am not sure). 

# 2. Potential Scope Beyond `view_img`

Other plotting utilities may rely on similar assumptions. If broader impact is confirmed, options include:

- Adding an optional `force_isotropic` parameter to `reorder_img`
- Refactoring `reorder_img` for clarity (current logic is difficult to follow)

If limited to HTML viewer path, fix can remain local to `load_bg_img`.

# 3. Efficiency Observation

When `bg_img is None`:

- Foreground is duplicated and processed twice.

Potential improvement:

- Avoid redundant preprocessing.
- Allow background-only viewing mode for structural inspection.

**Comment by michellewang** [comment]
I checked `plot_stat_map` and it seems the resampling is done correctly there:

```python
from nilearn.plotting import plot_stat_map
from nilearn import datasets

haxby_dataset = datasets.fetch_haxby()

plot_stat_map(
    haxby_dataset.anat[0],
    bg_img=None,
    cmap="gray",
    symmetric_cbar=False,
    threshold=None,
    vmax=250,
)
```

<img width="750" height="370" alt="Image" src="https://github.com/user-attachments/assets/a9d8350c-8c0a-4bb7-ab30-129b5f4b9d31" />

So probably this does not need to go beyond the scope of `view_img`
